src/browser_action/index.jsx

```diff
@@ -20,6 +20,12 @@ if (extractedProductJSON) {
   appProps.extractedProduct = JSON.parse(extractedProductJSON);
 }
 
+(async function checkOverflow() {
+  if (await browser.customizableUI.isWidgetInOverflow('shopping-testpilot_mozilla_org-browser-action')) {
+    document.getElementById('browser-action-app').classList.add('overflow');
```

Comment:
Hmmm, maybe we could put this into one of the React components, like in the state for `BrowserActionApp`? Not sure if that adds much to this.

Reply:
I tried to do this just now, but `<BrowserActionApp />` currently returns a `<React.Fragment>`, which doesn't accept a class name. Wrapping the whole thing in a `<div>` instead and adding the class to that is possible, but then the CSS also needs another layer instead of applying the `overflow` class directly to `#browser-action-app`:

```css
#browser-action-app > div {
  width: 320px;
}

#browser-action-app > div.overflow {
  width: 100%;
}
```

Also with the new Study Invitation screen (#274 ), I'd need that wrapper `div` in two places in BrowserActionApp instead of just one.

I also tried putting the class name directly on `<BrowserActionApp />` in `index.jsx`, but `ReactDOM.render` didn't like the `await` keyword needed for `browser.customizableUI.isWidgetInOverflow`.

I think for simplicity, I'll just leave this as is for now. Perhaps we can refactor this later if we ever implement the view router pattern you suggested [here](https://github.com/mozilla/price-wise/pull/278#discussion_r234744740)... The options I can think of/try right now seem more confusing and more prone to bugs.